modules/tinymce/src/core/main/ts/Rtc.ts

```diff
@@ -0,0 +1,296 @@
+    insertContent: (value, details) => insertHtmlAtCaret(editor, value, details),
+    getSelectedContent: (format, args) => getSelectedContentInternal(editor, format, args)
+  },
+  raw: {
+    getModel: () => Option.none()
+  }
+});
+
+const makeRtcAdaptor = (tinymceEditor: Editor, rtcEditor: RtcRuntimeApi): RtcAdaptor => {
+  const defaultVars = (vars: Record<string, string>) => Type.isObject(vars) ? vars : {};
+  const unsupported = Fun.die('Unimplemented feature for rtc');
+  const ignore = Fun.noop;
+  return {
+    undoManager: {
+      beforeChange: ignore,
+      addUndoLevel: unsupported,
+      undo: () => {
+        rtcEditor.undo();
+        return createDummyUndoLevel();
+      },
+      redo: () => {
+        rtcEditor.redo();
+        return createDummyUndoLevel();
+      },
+      clear: unsupported,
+      reset: unsupported,
+      hasUndo: () => rtcEditor.hasUndo(),
+      hasRedo: () => rtcEditor.hasRedo(),
+      transact: (_undoManager, _locks, fn) => {
+        rtcEditor.transact(fn);
+        return createDummyUndoLevel();
+      },
+      ignore: unsupported,
+      extra: unsupported
+    },
+    formatter: {
+      applyFormat: (name, vars, _node) => rtcEditor.applyFormat(name, defaultVars(vars)),
+      removeFormat: (name, vars, _node) => rtcEditor.removeFormat(name, defaultVars(vars)),
+      toggleFormat: (_formats, name, vars, _node) => rtcEditor.toggleFormat(name, defaultVars(vars)),
+    },
+    editor: {
+      getContent: (args, format) => {
+        if (isSupportedContentFormat(format)) {
+          const fragment = rtcEditor.getContent();
+          const serializer = Serializer({ inner: true });
+
+          FilterNode.filter(tinymceEditor.serializer.getNodeFilters(), tinymceEditor.serializer.getAttributeFilters(), fragment);
+
+          return serializer.serialize(fragment);
+        } else {
+          return makePlainAdaptor(tinymceEditor).editor.getContent(args, format);
+        }
+      },
+      getSelectedContent: (format, args) => {
+        if (isSupportedContentFormat(format)) {
+          const fragment = rtcEditor.getSelectedContent();
+          const serializer = Serializer({});
+          return serializer.serialize(fragment);
+        } else {
+          return makePlainAdaptor(tinymceEditor).editor.getSelectedContent(format, args);
+        }
+      },
+      setContent: (content, _args) => {
+        const fragment = isTreeNode(content) ? content : tinymceEditor.parser.parse(content, { isRootContent: true, insert: true });
+        rtcEditor.setContent(fragment);
+        return content;
+      },
+      insertContent: (value, _details) => {
+        const fragment = isTreeNode(value) ? value : tinymceEditor.parser.parse(value, { insert: true });
+        rtcEditor.insertContent(fragment);
+      }
+    },
+    raw: {
+      getModel: () => Option.some(rtcEditor.getRawModel())
+    }
+  };
+};
+
+export const isRtc = (editor) => Obj.has(editor.plugins, 'rtc');
```

Comment:
Types on the editor variable?

Reply:
Tell you what, when you have been developing in ReasonML for months you come back and forget how function definitions generally require type annotations

- [ ] add types